File: app/src/main/java/pl/mosenko/sunnypodlaskie/network/api/RxWeatherDataAPI.java
package pl.mosenko.sunnypodlaskie.network.api;

import io.reactivex.Observable;
import pl.mosenko.sunnypodlaskie.network.dto.WeatherDataDto;
import pl.mosenko.sunnypodlaskie.util.WeatherAPIKeyProvider;

/**
 * Created by syk on 13.05.17.
 */

public class RxWeatherDataAPI {

    private WeatherDataAPI weatherDataAPI;
    private WeatherAPIKeyProvider apiKeyProvider;

    public RxWeatherDataAPI(WeatherDataAPI weatherDataAPI, WeatherAPIKeyProvider apiKeyProvider) {
        this.weatherDataAPI = weatherDataAPI;
        this.apiKeyProvider = apiKeyProvider;
    }

    public Observable<WeatherDataDto> getCurrentWeatherData() {
        return weatherDataAPI.getCurrentWeatherData(apiKeyProvider.getDecodedAPIKey());
    }
}

File: app/src/main/java/pl/mosenko/sunnypodlaskie/network/api/RxWeatherDataAPI.kt
package pl.mosenko.sunnypodlaskie.network.api

import io.reactivex.Observable
import pl.mosenko.sunnypodlaskie.network.dto.WeatherDataDto
import pl.mosenko.sunnypodlaskie.util.WeatherAPIKeyProvider

/**
 * Created by syk on 13.05.17.
 */
class RxWeatherDataAPI(private val weatherDataAPI: WeatherDataAPI, private val apiKeyProvider: WeatherAPIKeyProvider) {
    fun getCurrentWeatherData(): Observable<WeatherDataDto> {
        return weatherDataAPI.getCurrentWeatherData(apiKeyProvider.getDecodedAPIKey())
    }
}
